### PR TITLE
[SYCL] Re-implement warning that checks number of arguments

### PR DESCRIPTION
**Pull request by Fznamznon**
GPU has limitation on size of kernel aguments, not number.

### CODE REVIEW

**Comment by bader** [comment]
I suggest we rename clang/test/SemaSYCL/num-args-overflow.cpp test.

File: clang/include/clang/Basic/DiagnosticSemaKinds.td [hunk 1]
```diff
@@ -10999,8 +10999,8 @@ def err_sycl_restrict : Error<
   "|use a const static or global variable that is neither zero-initialized "
   "nor constant-initialized"
   "}0">;
-def warn_sycl_kernel_too_many_args : Warning<
-  "kernel argument count (%0) exceeds supported maximum of %1 on GPU">,
+def warn_sycl_kernel_too_big_args : Warning<
+  "kernel arguments size (%0 bytes) exceeds supported maximum of %1 on GPU">,
```
Comment (premanandrao) [inline clang/include/clang/Basic/DiagnosticSemaKinds.td:11003]: Suggestion:
"size of kernel arguments (%0 bytes) exceeds supported maximum of %1 bytes on GPU"

**Review by premanandrao — COMMENTED** [review]
But otherwise, looks okay to me.

File: clang/test/SemaSYCL/args-size-overflow.cpp [hunk 2]
```diff
@@ -13,10 +13,10 @@ __attribute__((sycl_kernel)) void kernel(F KernelFunc) {
 template <typename Name, typename F>
 void parallel_for(F KernelFunc) {
 #ifdef GPU
-  // expected-warning@+8 {{kernel argument count (2001) exceeds supported maximum of 2000 on GPU}}
+  // expected-warning@+8 {{kernel arguments size (7994 bytes) exceeds supported maximum of 2048 on GPU}}
   // expected-note@+7 {{array elements and fields of a class/struct may be counted separately}}
```
Comment (premanandrao) [inline clang/test/SemaSYCL/args-size-overflow.cpp:17]: Also, the "note" message no longer makes sense as it does not make a difference due to decomposition.

**Review by elizabethandrews — COMMENTED** [review]
LGTM other than the unnecessary 'note'

File: clang/test/SemaSYCL/args-size-overflow.cpp (hunk 2 above)
Comment (Fznamznon) [reply clang/test/SemaSYCL/args-size-overflow.cpp]: Done.

File: clang/include/clang/Basic/DiagnosticSemaKinds.td (hunk 1 above)
Comment (Fznamznon) [reply clang/include/clang/Basic/DiagnosticSemaKinds.td]: Done.

**Review by premanandrao — APPROVED** [review]

**Pull request merged**